go-controller/pkg/ovn/zone_interconnect/interconnect.go

```diff
@@ -0,0 +1,535 @@
+ * ------------------------
+ * When network controller calls AddLocalZoneNode(ovn-worker)
+ *    -  A logical switch port - router port pair is created connecting the ovn_cluster_router
+ *       to the transit switch.
+ *    -  Node annotation - k8s.ovn.org/ovn-node-transit-switch-port-ifaddr value is used
+ *       as the logical router port address
+ *
+ * When network controller calls AddRemoteZoneNode(ovn-worker3)
+ *    - A logical switch port of type "remote" is created in OVN Northbound transit_switch
+ *      for the node ovn-worker3
+ *    - A static route {IPPrefix: "ovn-worker3_subnet", Nexthop: "ovn-worker3_transit_port_ip"} is
+ *      added in the ovn_cluster_router.
+ *    - For the default network, additional static route
+ *      {IPPrefix: "ovn-worker3_gw_router_port_host_ip", Nexthop: "ovn-worker3_transit_port_ip"} is
+ *      added in the ovn_cluster_router
+ *    - The corresponding port binding row in OVN Southbound DB for this logical port
+ *      is manually bound to the remote OVN Southbound DB Chassis "ovn-worker3"
+ *
+ * -----------------------------------------------------------------------------------------------------
+ * $ ovn-nbctl show ovn_cluster_router (on ovn-worker zone DB)
+ *   router ovn_cluster_router
+ *   ...
+ *   port rtots-ovn-worker
+ *      mac: "0a:58:a8:fe:00:08"
+ *      networks: ["168.254.0.8/16", "fd97::8/64"]
+ *
+ * $ ovn-nbctl show transit_switch
+ *     port tstor-ovn-worker
+ *        type: router
+ *        router-port: rtots-ovn-worker
+ *     port tstor-ovn-worker3
+ *        type: remote
+ *        addresses: ["0a:58:a8:fe:00:02 168.254.0.2/16 fd97::2/64"]
+ *
+ * $ ovn-nbctl lr-route-list ovn_cluster_router
+ *    IPv4 Routes
+ *    Route Table <main>:
+ *    ...
+ *    ...
+ *    10.244.0.0/24 (ovn-worker3 subnet)            168.254.0.2 (ovn-worker3 transit switch port ip) dst-ip
+ *    100.64.0.2/32 (ovn-worker3 gw router port ip) 168.254.0.2 dst-ip
+ *    ...
+ *    IPv6 Routes
+ *    Route Table <main>:
+ *    ...
+ *    ...
+ *    fd00:10:244:1::/64 (ovn-worker3 subnet)       fd97::2 (ovn-worker3 transit switch port ip) dst-ip
+ *    fd98::2 (ovn-worker3 gw router port ip)       fd97::2 dst-ip
+ *    ...
+ *
+ * $ ovn-sbctl show
+ *     ...
+ *     Chassis "c391c626-e1f0-4b1e-af0b-66f0807f9495"
+ *     hostname: ovn-worker3 (Its a remote chassis entry on which tstor-ovn-worker3 is bound)
+ *     Encap geneve
+ *         ip: "10.89.0.26"
+ *         options: {csum="true"}
+ *     Port_Binding tstor-ovn-worker3
+ *
+ * -----------------------------------------------------------------------------------------------------
+ *
+ * Note that the Chassis entry for each remote zone node is created by ZoneChassisHandler
+ *
+ */
+
+// ZoneInterconnectHandler creates the OVN resources required for interconnecting
+// multiple zones for a network (default or secondary layer 3)
+type ZoneInterconnectHandler struct {
+	// network which is inter-connected
+	util.NetInfo
```

Comment:
We shouldn't really be passing NetInfo around because it has an interface to modify the network/NAD relationship with which only the NAD and network controllers should be concerned about. I am working on that here: https://github.com/ovn-org/ovn-kubernetes/pull/3527

I will rebase but in case that PR get's in first you would only have to chang the type to BasicNetInfo